module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,67 @@
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+#include <algorithm>
+#include <string>
+
+std::string getErrorMessage(const ErrorCode& error){
+    switch(error){
+        case ErrorCode::Ok:
+        return "Ok";
+        
+        case ErrorCode::PasswordNeedsAtLeastNineCharacters:
+        return "Password needs at least nine characters";
+
+        case ErrorCode::PasswordNeedsAtLeastOneNumber:
+        return "Password needs at least one number";
+
+        case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter:
+        return "Password needs at least one special character";
+
+        case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter:
+        return "Password needs at least one upper case letter";
+
+        case ErrorCode::PasswordsDoesNotMatch:
+        return "Passwords does not match";
+        
+        default:
+        return "Unknown error";
+    
+    }
+    
+    
+}
+
+bool doesPasswordsMatch(const std::string& password1,const std::string& password2){
+    return password1==password2;
+}
+
+ErrorCode checkPasswordRules(const std::string& password){
+    if (password.length() < 9){
```

Comment:
Please avoid magic numbers
```suggestion
    constexpr int minPasswordLength = 9;
    if (password.length() < minPasswordLength){
```